drivers/mongodb/mongodb-springdata-v2-driver/src/main/java/io/mongock/driver/mongodb/springdata/v2/config/SpringDataMongoV2ContextBase.java

```diff
@@ -26,13 +26,19 @@ public ConnectionDriver connectionDriver(MongoTemplate mongoTemplate,
 
   protected abstract DRIVER buildDriver(MongoTemplate mongoTemplate,
                                         CONFIG config,
-                                        MongoDBConfiguration mongoDbConfig,
-                                        Optional<PlatformTransactionManager> txManagerOpt);
+                                        MongoDBConfiguration mongoDbConfig);
 
   private void setGenericDriverConfig(CONFIG config,
                                       Optional<PlatformTransactionManager> txManagerOpt,
                                       DRIVER driver) {
-    txManagerOpt.filter(tx -> config.getTransactionEnabled().orElse(true)).ifPresent(tx -> driver.enableTransaction());
+    if (config.getTransactional().isPresent()) {
+      if (config.getTransactional().get()) {
+        driver.enableTransaction();
```

Comment:
You can simplify it  with
`config.getTransactional().ifPresent(present-> driver.enableTransaction());`